### PR TITLE
fix: Reset the noir-gates-diff report on master

### PR DESCRIPTION
**Pull request by vezenovm**
# Description

## Problem\*

Resolves 

Opened this PR to test https://github.com/TomAFrench/noir-gates-diff/pull/3 after seeing this https://github.com/noir-lang/noir/pull/4848#issuecomment-2070241529

## Summary\*

The noir-gates-diff report became malformed on master as "main" was used as the program name for each program in `execution_success`. This PR references the last commit in this PR https://github.com/TomAFrench/noir-gates-diff/pull/3. That PR changes to using the correct name for each program in `execution_success` but also resets the report on `master` by comparing against a harcoded reference. That is why it is expected to see a massive gate diff report on this PR. The follow-up will revert the changes to accurately check against the reference.

## Documentation\*

Check one:
- [ ] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by vezenovm** [comment]
> Program	ACIR opcodes (+/-)	%	Circuit size (+/-)	%
> **hashmap**	+221,458 ❌	**+22145800.00%**	+563,008 ❌	**+56300800.00%**

We need to merge this PR with its garbage report data first to get the reference report on master to use all the correct package names. Then once this is merged we can followup with https://github.com/noir-lang/noir/pull/4891 which switches us back to using the actual reference report on master.

File: test_programs/execution_success/1_mul/src/main.nr [hunk 1]
```diff
@@ -1,6 +1,12 @@
 // Test unsafe integer multiplication with overflow: 12^8 = 429 981 696
 // The circuit should handle properly the growth of the bit size
 fn main(mut x: u32, y: u32, z: u32) {
+    let nine = x * x;
+    assert(nine == 9);
+
+    let sixteen = y * y;
+    assert(sixteen == 16);
```
Comment (jfecher) [inline test_programs/execution_success/1_mul/src/main.nr:8]: Was this change intended?

File: tooling/nargo_cli/src/cli/info_cmd.rs [hunk 2]
```diff
@@ -238,7 +238,7 @@ impl From<ProgramInfo> for Vec<Row> {
     fn from(program_info: ProgramInfo) -> Self {
         vecmap(program_info.functions, |function| {
             row![
-                Fm->format!("{}", program_info.name),
+                Fm->format!("{}", program_info.package_name),
```
Comment (jfecher) [inline tooling/nargo_cli/src/cli/info_cmd.rs:241]: Same here

File: test_programs/execution_success/1_mul/src/main.nr (hunk 1 above)
Comment (vezenovm) [reply test_programs/execution_success/1_mul/src/main.nr]: oops yup, just pushed

**Review by jfecher — APPROVED** [review]

**Comment by jfecher** [comment]
We'll blame this PR if anyone asks about performance regressions
